UnityProject/Assets/Scripts/HealthV2/Living/CirculatorySystem/BodyPartBlood.cs

```diff
@@ -8,192 +8,271 @@
+		public ReagentContainerBody BloodContainer = null;
 
-		public float BloodReagentStoreAmount => bloodReagentStoreAmount;
+		/// <summary>
+		/// The maximum size of the Blood Container
+		/// </summary>
+		public float BloodStoredMax => BloodContainer.MaxCapacity;
 
-		[Tooltip("Can we store any blood reagent?")]
-		public  float bloodReagentStoredMax = 0.5f;
+		/// <summary>
+		/// The amount (in moles) of required reagent (eg oxygen) this body part needs consume each tick.
+		/// </summary>
+		[Tooltip("How much (in moles) blood reagent (eg oxygen) does this need each tick?")]
+		[SerializeField] private float bloodReagentConsumed = 0.00002f;
 
-		private float BloodDamageLow = 0;
+		[Tooltip("How much blood reagent does this request per blood pump event?")]
+		[SerializeField] private float bloodThroughput = 0.015f; //This will need to be reworked when heartrate gets finished
+		/// <summary>
+		/// The amount of blood ReagentMix this body part will remove and add each blood pump event
+		/// Essentially controls the rate of blood flow through the organ
+		/// </summary>
+		public float BloodThroughput => bloodThroughput;
 
-		//private float bloodReagentStored = 0;
+		/// <summary>
+		/// The nutriment reagent that this part consumes in order to perform tasks
+		/// </summary>
+		[Tooltip("What does this live off?")]
+		[SerializeField] public Reagent Nutriment;
 
-		public float NutrimentConsumption = 0.02f;
+		/// <summary>
+		/// The amount of of nutriment to consumed each tick as part of passive metabolism
+		/// </summary>
+		[Tooltip("How much nutriment does this passively consume to each tick?")]
 		public float NutrimentPassiveConsumption = 0.001f;
 
+		/// <summary>
+		/// The amount of of nutriment to consume in order to perform work, eg heal damage or replenish blood supply
+		/// </summary>
+		[Tooltip("How much nutriment does this consume to perform work?")]
+		public float NutrimentConsumption = 0.02f;
 
-		public ReagentContainerBody BloodContainer = null;
-
-		[SerializeField] [Tooltip("What does this live off?")]
-		public Reagent Nutriment;
-
-		public event Action ModifierChange;
-
-		public float TotalModified = 1;
-
-		public List<Modifier> AppliedModifiers = new List<Modifier>();
-
-		public void UpdateMultiplier()
+		/// <summary>
+		/// Initializes the body part as part of the circulatory system
+		/// </summary>
+		public void BloodInitialise()
 		{
-			TotalModified = 1;
-			foreach (var Modifier in AppliedModifiers)
+			BloodContainer = this.GetComponent<ReagentContainerBody>();
+			if (BloodContainer.ContentsSet == false)
 			{
-				TotalModified *= Mathf.Max(0, Modifier.Multiplier);
+				if (isBloodCirculated)
+				{
+					HealthMaster.CirculatorySystem.ReadyBloodPool.TransferTo(BloodContainer.CurrentReagentMix, BloodStoredMax);
+				}
+				BloodContainer.ContentsSet = true;
+			}
+			if (bloodType == null)
+			{
+				bloodType = HealthMaster.CirculatorySystem.BloodType;
```

Comment:
humm What's the desired outcome if you attach plasma man arm to you,
I would expect it to work as long as you have plasma in your bloodstream,
if you have plasma in your bloodstream then it takes it from the blood and goes this is good,
I suppose organs don't care about the blood in them, they just care if it has the gas they require/reagent

Reply:
Actually, as I have it implemented the default body types only take required reagent from the blood type they are assigned (as part of ConsumeReagents()), I was thinking we wanted implant rejection, so you would need plasma men blood in your system.  Of course, any body part can override how they use reagents if the body part should be generic, or part of the surgery procedure could assign the blood type to match whoever they are grafting it to.  Or I could remove this check altogether, but then you would be able to get transfusions from any blood type.

Reply:
yeah, so what is the point of blood type then other than lungs for like balancing chemicals right?

Reply:
Umm, I'm not sure I understand.  The point of the blood type is to make it very difficult to use other people's body parts in your body, and I think having the lungs refuse to oxygenate foreign blood, the kidney removing foreign blood, and your body parts not pulling oxygen from foreign blood, along with the part only working off of foreign blood simulates that quite well.  To not need constant external help you would need a transfusion of 5L of foreign blood, foreign lungs, remove your kidneys, a second heart, and you would still run into problems of your blood being 50% dilute.  I haven't tested any of this because non-human blood isn't done yet, so the body may just fail even then.

If that is NOT what we want, if we want foreign body parts to be usable freely, then yes, blood types are pointless and the code should be rewritten entirely.  Frankly, at that point, body parts should not have blood in them at all, and only reagents, and the heart should pump reagents to them, and blood would just be a number that determines how effective the heart and lungs are, and does oxy damage to the whole body when its low.

Reply:
Okay, I had misunderstood something pretty fundamental there, blood was still generic, ie there wasn't a different reagent for each kind of blood.  This has been fixed now, BloodType is the actual reagent being pumped, BloodType.Type is that blood's blood type (A, B, O, etc).

Reply:
I kinda like the idea of organ rejection and making cross specie implants not ideal or harder to make them work although leaving the option and mechanics so they are possible.